pkg/cephfs/driver.go

```diff
@@ -147,20 +147,30 @@ func (fs *Driver) Run(conf *util.Config, cachePersister util.CachePersister) {
 		klog.Fatalln("failed to initialize CSI driver")
 	}
 
-	fs.cd.AddControllerServiceCapabilities([]csi.ControllerServiceCapability_RPC_Type{
-		csi.ControllerServiceCapability_RPC_CREATE_DELETE_VOLUME,
-	})
-
-	fs.cd.AddVolumeCapabilityAccessModes([]csi.VolumeCapability_AccessMode_Mode{
-		csi.VolumeCapability_AccessMode_MULTI_NODE_MULTI_WRITER,
-	})
-
+	if conf.IsControllerServer || !(conf.IsControllerServer && conf.IsNodeServer) {
```

Comment:
If isNS is `true`and isCS is `false` we would still go ahead and add the capability information as the `!(isCS && isNS)` evaluates to true, similarly in line 170, if only one of the services are true, then both services would be started.

I think the code intends to do the following in line 150,
```
if isCS || !isNS
```

and in line 170 this,
```
if !isCS && !isNS
```

Reply:
yes, updated code now